src/libsync/httplogger.cpp

```diff
@@ -42,47 +44,39 @@ bool isTextBody(const QString &s)
     return regexp.match(s).hasMatch();
 }
 
-void logHttp(const QByteArray &verb, const QString &url, const QByteArray &id, const QString &contentType, const QList<QNetworkReply::RawHeaderPair> &header, QIODevice *device, const nanoseconds &duration = {})
+void logHttp(const QByteArray &verb, const QString &url, const QByteArray &id, const QString &contentType, QJsonObject &&header, QIODevice *device,
+    const nanoseconds &duration = {})
 {
+    static const bool redact = !qEnvironmentVariableIsSet("OWNCLOUD_HTTPLOGGER_NO_REDACT");
     const auto reply = qobject_cast<QNetworkReply *>(device);
     const auto contentLength = device ? device->size() : 0;
-    QString msg;
-    QDebug stream(&msg);
-    stream.nospace().noquote();
-    stream << id << ": ";
-    if (!reply) {
-        stream << "Request: ";
-    } else {
-        stream << "Response: ";
+
+    if (redact) {
+        const QString authKey = QStringLiteral("Authorization");
+        const QString auth = header.value(authKey).toString();
+        header.insert(authKey, auth.startsWith(QStringLiteral("Bearer ")) ? QStringLiteral("Bearer [redacted]") : QStringLiteral("Basic [redacted]"));
     }
-    stream << verb;
+
+    QJsonObject info{{QStringLiteral("method"), QString::fromUtf8(verb)}, {QStringLiteral("id"), QString::fromUtf8(id)}, {QStringLiteral("url"), url}};
+
     if (reply) {
-        stream << " " << reply->attribute(QNetworkRequest::HttpStatusCodeAttribute).toInt() << " (";
+        QString durationString;
+        QDebug(&durationString).nospace() << duration;
+        QJsonObject replyInfo{{QStringLiteral("status"), reply->attribute(QNetworkRequest::HttpStatusCodeAttribute).toInt()},
+            {QStringLiteral("cached"), reply->attribute(QNetworkRequest::SourceIsFromCacheAttribute).toBool()},
+            // downcast to int, this is json
```

Comment:
We could also use string? but int should be more than sufficient.

Reply:
No but gcc complained, I guess because json does not support int64

Reply:
There seems to be some weird 53-bit limit in the spec and various services seem to implement different workarounds (some actually send strings along with the downcast int values). This should be fine, it's rather unlikely we'll overflow.